### PR TITLE
Feature/prettier

### LINKED ISSUE #203: Thoughts on using prettier for code?
Prettier is becoming more standard across Typescript repos and enforces an acceptable style. Thoughts on introducing it to the code base? I saw that tslint was added somewhat recently.

### PR DESCRIPTION
**Pull request by sheldonkwok**
Adds prettier to project setup and run on the current codebase 
Closes #203

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
+    "format": "prettier --loglevel error --write './src/**/*.ts'",
```
Comment (drubin) [inline package.json]: Is `prettier ` this not missing as a dev dependency? 

Also is there a way to ensure the format is valid as part of CI? I think it should be as simple as adding `npm run format` but you might need to validate it works.

Comment (sheldonkwok) [reply package.json]: Oops I accidentally rebased it out. 

Just added a `prettier --check` to `lint`

**Comment by brendandburns** [comment]
This looks good to me. @drubin or @itowlson any other comments before we merge?

**Comment by itowlson** [comment]
Looks fine to me too; let's give it a whirl.

**Comment by drubin** [comment]
/assign @brendandburns 

You can approve when you are happy.

File: src/config_test.ts [hunk 2]
```diff
@@ -662,14 +666,16 @@ describe('KubeConfig', () => {
...
                             'cmd-path': 'non-existent-command',
                         },
                     },
-                } as User);
+                } as User,
+            );
             const opts = {} as requestlib.Options;
             expect(() => config.applyToRequest(opts)).to.throw(
-                /Failed to refresh token/);
+                /Failed to refresh token/
```
Comment (brendandburns) [inline src/config_test.ts]:
```suggestion
                /Failed to refresh token/,
```

**Comment by brendandburns** [comment]
@sheldonkwok  There were rebase merge conflicts, I tried to resolve them, but I ended up breaking tslint. I think my suggestion fixes it... Anyway, please re-sync and then I'll approve/merge

Apologies!

**Comment by sheldonkwok** [comment]
@brendanburns I just rebased with master

**Pull request merged**